### PR TITLE
Support for RTL text plugin 0.3.0

### LINKED ISSUE #4850: mapbox-gl-rtl-text v0.3.0 support
We are loading maplibre-gl 4.7.1 dynamically from esm which works well.

This allows us to use the mapbox-gl-rtl-text@0.2.3 loaded from the unpkg cdn.

When we are trying to update the plugin to the latest version https://unpkg.com/@mapbox/mapbox-gl-rtl-text@0.3.0/dist/mapbox-gl-rtl-text.js we hit an error on plugin registration and arabix text is not displayed.

image

"RTL Text Plugin failed to import scripts from https://unpkg.com/@mapbox/mapbox-gl-rtl-text@0.3.0/dist/mapbox-gl-rtl-text.js"

This is not a big issue since 0.2.3 version is working as expected though I would like to understand the nature of the issue.

```js
  // Create promise to load maplibre from esm.sh
  if (!promise) promise = new Promise(async resolve => {

    try {

      const mod = await import('https://esm.sh/maplibre-gl@4.7.1')

      maplibregl = mod.default

      // Avoid loading RTL Text Plugin twice, else it will error
      if (!['deferred', 'loaded'].includes(maplibregl.getRTLTextPluginStatus())) {
        maplibregl.setRTLTextPlugin(
          'https://unpkg.com/@mapbox/mapbox-gl-rtl-text@0.2.3/mapbox-gl-rtl-text.min.js',
          true // Lazy load the plugin
        );
      }

      resolve()

    } catch {

      console.error('Failed to load maplibre-gl library.')
      resolve()
    }
  })
```

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

- #4850
- This allows loading the script in a async form as 0.3.0 plugin is now loading a wasm in ansyc manner.
- This moves all the RTL logic into the relevant class and reduce API surface.
- Remove some unwanted console log in tests
- This code supports both 0.2.3 and 0.3.0 as far as I've tested.

Fixes #4850

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/source/rtl_text_plugin_worker.test.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        const workerResult: PluginState = await await rtlWorkerPlugin.syncState(mockMessage, jest.fn());
+
+        expect(rtlWorkerPlugin.getRTLTextPluginStatus()).toBe('loaded');
+        expect(rtlWorkerPlugin.pluginURL).toBe(originalUrl);
+
+        expect(workerResult.pluginStatus).toBe('loaded');
+        expect(workerResult.pluginURL).toBe(originalUrl);
+    });
+
+    test('should do a full cycle of rtl loading synchronously', async () => {
```
Comment (HarelM) [inline src/source/rtl_text_plugin_worker.test.ts:62]: This is how 0.2.3 works

File: src/source/rtl_text_plugin_worker.test.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            pluginStatus: 'loading'
+        }, loadScriptsMock);
+
+        expect(rtlWorkerPlugin.getRTLTextPluginStatus()).toBe('loaded');
+        expect(rtlWorkerPlugin.pluginURL).toBe(originalUrl);
+        expect(workerResult.pluginStatus).toBe('loaded');
+        expect(workerResult.pluginURL).toBe(originalUrl);
+    });
+
+    test('should do a full cycle of rtl loading asynchronously', async () => {
```
Comment (HarelM) [inline src/source/rtl_text_plugin_worker.test.ts:83]: This is how 0.3.0 works.
I've added both tests to make sure both option will be supported in the future.

File: src/source/geojson_source.test.ts [hunk 3]
```diff
@@ -302,6 +302,7 @@ describe('GeoJSONSource#update', () => {
     }));
 
     test('forwards Supercluster options with worker request, ignore max zoom of source', () => new Promise<void>(done => {
+        jest.spyOn(console, 'warn').mockImplementation(() => {});
```
Comment (ibesora) [inline src/source/geojson_source.test.ts:305]: This seems unrelated to this PR, maybe we should move it to another cleanup PR?

File: src/ui/camera.test.ts [hunk 4]
```diff
@@ -3502,7 +3502,6 @@ describe('#flyTo globe projection', () => {
             });
 
             camera.on('moveend', () => {
-                console.log(leastZoom);
```
Comment (ibesora) [inline src/ui/camera.test.ts:3505]: Same as the previous message. I'd leave only the RTL text plugin changes to this PR and move all these things to a test cleanup PR

**Review by ibesora — APPROVED** [review]
There are a bunch of changes that seem unrelated to what's being done in this PR which I'd move to another PR but feel free to ignore
Since examples and tests all work, I'd say this is ready to go!
Good job!

File: src/source/geojson_source.test.ts (hunk 3 above)
Comment (HarelM) [reply src/source/geojson_source.test.ts]: Reverted

File: src/ui/camera.test.ts (hunk 4 above)
Comment (HarelM) [reply src/ui/camera.test.ts]: Reverted
